pretyping/inductiveops.ml

```diff
@@ -758,8 +758,11 @@ let type_of_projection_knowing_arg env sigma p c ty =
    syntactic conditions *)
 
 let control_only_guard env sigma c =
+  let c = Evarutil.nf_evar sigma c in
   let check_fix_cofix e c =
-    match kind (EConstr.to_constr sigma c) with
+    (** [c] has already been normalized upfront *)
+    let c = EConstr.Unsafe.to_constr c in
```

Comment:
Don't we have a safe method to do this?

Reply:
As witnessed by #7286, this is hopeless, the argument might contain unresolved evars.